components/brave_wallet/browser/ethereum_keyring.cc

```diff
@@ -128,4 +129,18 @@ std::string EthereumKeyring::GetAddressInternal(HDKeyBase* hd_key_base) const {
   return addr.ToChecksumAddress();
 }
 
+bool EthereumKeyring::GetPublicKeyFromX25519_XSalsa20_Poly1305(
+    const std::string& address,
+    std::string* key) {
+  HDKey* hd_key = static_cast<HDKey*>(GetHDKeyFromAddress(address));
+  if (!hd_key)
+    return false;
+  const std::vector<uint8_t> public_key =
+      hd_key->GetPublicKeyFromX25519_XSalsa20_Poly1305();
+  if (public_key.size() != 32)
```

Comment:
It is better to write it as `!public_key.size()` since we don't export the size from hd_key and it is easy to get it wrong. Or we can do `absl::optional`

Reply:
Updated here https://github.com/brave/brave-core/pull/12517/files#diff-88cce901212dd08dfe99800009c4bc7e64f02ad0252a5cf1fb71d29b0689b326R140